### PR TITLE
feat: check that organizations match when creating catalogue item

### PR DESCRIPTION
**Pull request by opqdonut**
for #1893

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## API
- [x] API is backwards compatible or completely new

## Documentation
- [x] update changelog if necessary

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: src/clj/rems/api/services/catalogue.clj [hunk 1]
```diff
@@ -17,8 +17,21 @@
...
                                        :items [id]})}))
 
+(defn- organization-mismatch-error [{:keys [organization form resid wfid]}]
+  (let [mismatches (merge (when (not= organization (:form/organization (form/get-form-template form)))
+                            {:form form})
+                          (when (not= organization (:organization (resource/get-resource resid)))
+                            {:resource resid})
+                          (when (not= organization (:organization (workflow/get-workflow wfid)))
+                            {:workflow wfid}))]
+    (when (not (empty? mismatches))
```
Comment (Macroz) [inline src/clj/rems/api/services/catalogue.clj]: I'd prefer just `(when (seq mismatches) ...)` here. It's currently the most popular option in the codebase?

File: src/clj/rems/db/test_data.clj [hunk 2]
```diff
@@ -223,10 +223,10 @@
                                      :infourl (get infourl lang)}]))
         result (with-user actor
                  (catalogue/create-catalogue-item!
-                  {:resid (or resource-id (create-resource! {}))
-                   :form (or form-id (create-form! {}))
+                  {:resid (or resource-id (create-resource! {:organization organization}))
```
Comment (Macroz) [inline src/clj/rems/db/test_data.clj]: Should these handle still the default i.e. no organization? Or is the intended direction to always require it?

File: test/clj/rems/api/test_catalogue_items.clj [hunk 3]
```diff
@@ -58,6 +58,13 @@
                 (is (response-is-not-found? response))
                 (is (= "application/json" (get-in response [:headers "Content-Type"])))))))
 
+        (testing "create with mismatched organization"
+          (let [data (create-catalogue-item "owner" "nbn")]
+            (is (not (:success data)))
+            (is (= [{:type "t.administration.errors/organization-mismatch"
+                     :form form-id :resource res-id :workflow wf-id}]
```
Comment (Macroz) [inline test/clj/rems/api/test_catalogue_items.clj]: Should it show the organizations here?

File: src/clj/rems/api/services/catalogue.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/api/services/catalogue.clj]: somehow `empty?` feels more natural for maps. for vectors & lists `seq` is natural

File: src/clj/rems/db/test_data.clj (hunk 2 above)
Comment (opqdonut) [reply src/clj/rems/db/test_data.clj]: if organization is nil, the defaulting inside create-resource!, create-form! etc. takes over. they all default to `"default"`

File: test/clj/rems/api/test_catalogue_items.clj (hunk 3 above)
Comment (opqdonut) [reply test/clj/rems/api/test_catalogue_items.clj]: hmm, why not. any suggestions for the structure of the error? two ideas:
```
{:form {:id form-id :organization "foo"} :resource {:id res-id :organization "boing"}}
{:form-organization "foo" :resource-organization "boing"}
```

File: src/clj/rems/api/services/catalogue.clj (hunk 1 above)
Comment (Macroz) [reply src/clj/rems/api/services/catalogue.clj]: Hmm, is `(when-let [mismatches ...] ...)` possible?

File: test/clj/rems/api/test_catalogue_items.clj (hunk 3 above)
Comment (Macroz) [reply test/clj/rems/api/test_catalogue_items.clj]: I think the first one makes more sense.

**Pull request merged**